src/core/packages/capabilities/server-internal/src/routes/resolve_capabilities.ts

```diff
@@ -35,7 +35,7 @@ export function registerCapabilitiesRoutes(router: IRouter, resolver: Capabiliti
             schema.string({
               validate: (appName) => {
                 if (!applicationIdRegexp.test(appName)) {
-                  return 'Invalid application id';
+                  return `Invalid application id: ${appName}`;
```

Comment:
Since users can inadvertently inject data into our logs, I think it would be good to limit what we log to the first 20 characters just to avoid someone sending a big payload.